shell/platform/windows/angle_surface_manager.cc

```diff
@@ -109,34 +82,76 @@ bool AngleSurfaceManager::Initialize() {
       d3d9_display_attributes,
   };
 
+  PFNEGLGETPLATFORMDISPLAYEXTPROC eglGetPlatformDisplayEXT =
+      reinterpret_cast<PFNEGLGETPLATFORMDISPLAYEXTPROC>(
+          eglGetProcAddress("eglGetPlatformDisplayEXT"));
+  if (!eglGetPlatformDisplayEXT) {
+    LogEglError("eglGetPlatformDisplayEXT not available");
+    return false;
+  }
+
   // Attempt to initialize ANGLE's renderer in order of: D3D11, D3D11 Feature
   // Level 9_3, D3D11 WARP and finally D3D9.
   for (auto config : display_attributes_configs) {
-    if (InitializeEGL(config)) {
-      break;
+    // We should only log if we're at the last attempt with no more fallbacks.
+    bool should_log = (config == display_attributes_configs.back());
+
+    egl_display_ = eglGetPlatformDisplayEXT(EGL_PLATFORM_ANGLE_ANGLE,
+                                            EGL_DEFAULT_DISPLAY, config);
+
+    if (egl_display_ == EGL_NO_DISPLAY) {
+      if (should_log) {
+        LogEglError("Failed to get a compatible EGLdisplay");
+      }
+      continue;
+    }
+
+    if (eglInitialize(egl_display_, nullptr, nullptr) == EGL_FALSE) {
+      if (should_log) {
+        LogEglError("Failed to initialize EGL via ANGLE");
+      }
+      continue;
     }
+
+    break;
```

Comment:
Optional: I think the old loop was easier to understand than this bare `break`; what about keeping the separate method for the body, and passing in `eglGetPlatformDisplayEXT` and `should_log` as parameters?

(On the other hand, when the logic is something like `succeeded = true; break;` that may resolve my readability concern.)